### PR TITLE
unify progress bar for datasets

### PR DESCRIPTION
**Pull request by rijobro**
unify arguments for progress bar in datasets.

### CODE REVIEW

**Review by wyli — APPROVED** [review]
Thanks!

File: monai/data/dataset.py [hunk 1]
```diff
@@ -51,14 +51,16 @@ class Dataset(_TorchDataset):
          },                           },                           }]
     """
 
-    def __init__(self, data: Sequence, transform: Optional[Callable] = None) -> None:
+    def __init__(self, data: Sequence, transform: Optional[Callable] = None, progress: bool = True) -> None:
```
Comment (wyli) [inline monai/data/dataset.py:54]: perhaps the `progress` is not a fundamental property of a `Dataset`, so it shouldn't be in this base class, what do you think? for a user of the `Dataset` API, it's difficult to understand this option without looking at the cache dataset and some of the concrete implementations... 

@Nic-Ma @rijobro   sorry I should have had this discussion earlier during my review, somehow I missed it...

Comment (rijobro) [reply monai/data/dataset.py:54]: I think you're right, sorry about that. A couple of choices:

1. New middleman class. `DatasetWithProgress`. `PersistentDataset` and `CacheDataset` would inherit from it.
2. Update current documentation for base class: `progress: whether to display a progress bar **(if relevant)**.`
3. Revert this PR.

Comment (wyli) [reply monai/data/dataset.py:54]: thanks, I vote for option 1... @Nic-Ma thoughts?

Comment (Nic-Ma) [reply monai/data/dataset.py:54]: I agree with you guys, `progress` should not be in the base class.

Thanks.

**Pull request merged**